### PR TITLE
add the docs on tcp traffic mirror analysis

### PR DESCRIPTION
**Pull request by AnastasiaTWW**

### CODE REVIEW

File: docs/4.10/installation/oob/tcp-traffic-mirror/configuration.md [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+  ListenAddress: :8080
+```
+
+## GoreplayMiddleware.Enabled (required)
+
+Controls whether the Wallarm node is enabled for TCP traffic mirror analysis.
+
+## GoreplayMiddleware.Goreplay.Filter
+
+Specifies a network interface to capture traffic from. If no value is specified, it captures traffic from all network interfaces on the instance. Example configuration:
```
Comment (w-kerman) [inline docs/4.10/installation/oob/tcp-traffic-mirror/configuration.md:59]: > network interface and port

One can make a filter like "eth0:80". Or even ":80" for intercepting specific port on all interfaces.

File: docs/4.10/installation/oob/tcp-traffic-mirror/configuration.md [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+
+To check network interfaces available on the host, run:
+
+```
+ip link show command
+```
+
+## GoreplayMiddleware.Goreplay.ExtraArgs
+
+Specifies arguments for what traffic to capture:
```
Comment (w-kerman) [inline docs/4.10/installation/oob/tcp-traffic-mirror/configuration.md:77]: Rather, "specifies extra arguments passed to goreplay".

There are a lot of other thing you can do with it, listed in goreplay docs: https://github.com/buger/goreplay/blob/master/docs/Request-filtering.md

File: docs/4.10/installation/oob/tcp-traffic-mirror/configuration.md [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+
+Controls whether to parse mirrored responses. This enables Wallarm features that rely on response data, such as [vulnerability detection](../../../about-wallarm/detecting-vulnerabilities.md) and [API discovery](../../../api-discovery/overview.md).
+
+By default, `true`.
+
+Ensure response mirroring is configured in your environment to the target instance with the Wallarm node.
+
+## GoreplayMiddleware.Middleware.ResponseTimeout
+
+Specifies the maximum time to wait for a response in seconds. If a response is not received within this time, the Wallarm processes stop waiting does not parse the corresponding response.
```
Comment (w-kerman) [inline docs/4.10/installation/oob/tcp-traffic-mirror/configuration.md:120]: Not necessarily in seconds.
Might be in any other unit, like "1m" or "500ms"

Also, the sentence "processes stop waiting does not parse" seems to lack something.

File: docs/4.10/installation/oob/tcp-traffic-mirror/configuration.md [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+
+Log level, can be `debug`, `info`, `warn`, `error`, `fatal`.
+
+Default: `info`.
+
+## Log.LogFile
+
+Specifies the destination for log output. Options are `stdout`, `stderr`, or a path to a log file.
+
+Default: `/opt/wallarm/var/log/wallarm/go-node.log`.
```
Comment (w-kerman) [inline docs/4.10/installation/oob/tcp-traffic-mirror/configuration.md:233]: Not exactly. It's `stderr` by default. But in AIO, the node is run by supervisord, and it redirects stderr into that file.

File: docs/4.10/installation/oob/tcp-traffic-mirror/deployment.md [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+
+This solution operates in out-of-band (OOB) mode, capturing mirrored TCP traffic directly from the network interface, independent of web servers like NGINX. The solution uses Goreplay to capture the traffic and handle encapsulations (e.g., VLAN, VXLAN). The captured traffic is then parsed, reassembled, and analyzed for threats by the Wallarm services.
+
+It seamlessly swaps between multiple traffic sources. When multiple data sources are directed to a traffic mirror target (the instance with the Wallarm node), the solution efficiently collects and decrypts traffic, allowing flexible monitoring without disruption. By reassembling TCP traffic, the Wallarm node ensures accurate threat detection and analysis, ensuring no suspicious activity is missed.
+
+Additionally, the solution supports response mirror parsing, allowing non-interrupted traffic flow while providing Wallarm features that rely on response data. These features include vulnerability discovery, API discovery, and more.
+
+## Requirements
+
+* Understanding of TCP traffic mirroring flow and the specifics of your environment to configure the solution effectively.
```
Comment (apogorelovwallarm) [inline docs/4.10/installation/oob/tcp-traffic-mirror/deployment.md:23]: do we really need to tell the user they are required to have the "understanding of the specifics of their environment"?

File: docs/4.10/installation/oob/tcp-traffic-mirror/deployment.md [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+
+    === "x86_64 version"
+        ```
+        sudo ./aionext-0.2.2.x86_64.sh -- --help
+        ```
+    === "ARM64 version"
+        ```
+        sudo ./aionext-0.2.2.aarch64.sh -- --help
+        ```
+* You can also run the installer in an **interactive** mode by choosing the `tcp-capture` mode in the 1st step:
```
Comment (apogorelovwallarm) [inline docs/4.10/installation/oob/tcp-traffic-mirror/deployment.md:210]:
```suggestion
* You can also run the installer in an **interactive** mode and choose the `tcp-capture` mode in the 1st step:
```

File: docs/4.10/installation/oob/tcp-traffic-mirror/configuration.md [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+```yaml
+Version: 1
+
+GoreplayMiddleware:
+  Enabled: true
+  Goreplay:
+    Filter: <your network interface, i.e. "lo:" or "enp7s0:">
+    ExtraArgs:
+      - ...
+      - ...
```
Comment (apogorelovwallarm) [inline docs/4.10/installation/oob/tcp-traffic-mirror/configuration.md:16]: other config values in this example imitate real-life values. I suggest just setting this to something like "enp7s0:" and i.e. 
```
      - -input-raw-engine
      - vxlan

File: docs/4.10/installation/oob/tcp-traffic-mirror/deployment.md [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+    Restart the Wallarm service:
+
+    ```
+    sydo systemctl restart wallarm
+    ```
+
+    Logs are written to `/opt/wallarm/var/log/wallarm/go-node.log` by default. You can read them there.
+* Standard logs of the filtering node such as whether the data is sent to the Wallarm Cloud, detected attacks, etc. are located in the directory `/opt/wallarm/var/log/wallarm`.
+
+## Installer launch options
```
Comment (apogorelovwallarm) [inline docs/4.10/installation/oob/tcp-traffic-mirror/deployment.md:198]: where's the upgrade instructions?

File: docs/4.10/installation/oob/tcp-traffic-mirror/configuration.md [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+          Wallarm:
+            ApplicationId: 2
+        - Route: "/testing"
+          Wallarm:
+            Mode: off
+  Wallarm:
+    WallarmDirPath: /etc/wallarm
+    RealIpHeader: "x-real-ip"
+  WallarmExporter:
+    Address: "127.0.0.1:3313"
```
[the remark this answers is not in the document]
Comment (apogorelovwallarm) [reply docs/4.10/installation/oob/tcp-traffic-mirror/configuration.md:37]: > every remaining non-important config parameter

and to get a list of those just run the go-node with debug log level and it'll spit out its whole config

File: docs/latest/installation/oob/tcp-traffic-mirror/configuration.md [hunk 10]
```diff
@@ -0,0 +1,354 @@
...
+
+Default: `true`.
+
+### GoreplayMiddleware.WallarmExporter.Address
+
+Sets the address for the postanalytics service which handles statistical request analysis in Wallarm's request processing. Typically, you do not need to modify this parameter.
+
+Default: `127.0.0.1:3313`.
+
+### GoreplayMiddleware.WallarmExporter.Enabled
```
Comment (w-kerman) [inline docs/latest/installation/oob/tcp-traffic-mirror/configuration.md:308]: Let's remove mentions of this completely

File: docs/latest/installation/oob/tcp-traffic-mirror/deployment.md [hunk 11]
```diff
@@ -0,0 +1,288 @@
...
+
+Create the `wallarm-node-conf.yaml` file on the instance. The solution requires proper configuration to identify the network interface and the traffic format (e.g., VLAN, VXLAN). The minimal content should include the following settings:
+
+```yaml
+Version: 1
+GoreplayMiddleware:
+  Enabled: true
+  Goreplay:
+    Filter: "enp7s0:"
+    ExtraArgs:
```
Comment (w-kerman) [inline docs/latest/installation/oob/tcp-traffic-mirror/deployment.md:83]: I don't think this should be under "minimal content". Just don't supply extra args here.

File: docs/latest/installation/oob/tcp-traffic-mirror/deployment.md [hunk 12]
```diff
@@ -0,0 +1,290 @@
...
+Wallarm suggests installations for the following processors:
+
+* x86_64
+* ARM64
+
+To download Wallarm installation script and make it executable, use the following commands:
+
+=== "x86_64 version"
+    ```bash
+    curl -O https://meganode.wallarm.com/next/aionext-0.3.0.x86_64.sh
```
Comment (apogorelovwallarm) [inline docs/latest/installation/oob/tcp-traffic-mirror/deployment.md:65]: do not forget to update this to the latest version before the docs release

File: docs/latest/installation/oob/tcp-traffic-mirror/deployment.md [hunk 13]
```diff
@@ -0,0 +1,290 @@
...
+  extra_args:
+      - -input-raw-engine
+      - vxlan
+```
+
+In the [article](configuration.md), you will find the list of more supported configuration parameters.
+
+### Setting the mode (required)
+
+It is required to specify the `tcp-capture` mode in the corresponding chart value to run the solution for the TCP traffic mirror analysis.
```
Comment (apogorelovwallarm) [inline docs/latest/installation/oob/tcp-traffic-mirror/deployment.md:94]: > chart value

config parameter?

File: docs/latest/installation/oob/tcp-traffic-mirror/configuration.md [hunk 14]
```diff
@@ -0,0 +1,341 @@
...
+  extra_args:
+    - -input-raw-engine
+    - vxlan
+
+route_config:
+  wallarm_application: 10
+  routes:
+    - route: /example/api/v1
+      wallarm_mode: off
+      wallarm_application: 1
```
Comment (apogorelovwallarm) [inline docs/latest/installation/oob/tcp-traffic-mirror/configuration.md:23]: is there a point in setting wallarm_application for the route where wallarm_mode is off? Probably no

File: docs/latest/installation/oob/tcp-traffic-mirror/configuration.md [hunk 15]
```diff
@@ -0,0 +1,341 @@
...
+version: 2
+
+goreplay:
+  filter: 'eth0:'
+```
+
+To check network interfaces available on the host, run:
+
+```
+ip link show command
```
Comment (apogorelovwallarm) [inline docs/latest/installation/oob/tcp-traffic-mirror/configuration.md:58]: firstly - the `command` word shouldn't be there at all.
Secondly, let's use `ip addr` command instead, as it additionally shows the interface IP address, which might be convenient

**Pull request merged**